### PR TITLE
ShellCheck for various helper bash scripts

### PR DESCRIPTION
**Pull request by seberm**
This PR adds shellcheck lint to helper bash scripts, for more info please see:
- https://github.com/fort-nix/nix-bitcoin/issues/528

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -11,7 +12,7 @@ task:
     - name: modules_test
       container:
         # Use Nix release 2.3.x, the default on NixOS stable
```
Comment (jonasnick) [inline .cirrus.yml:14]: This comment needs to be updated.

File: test/ci/shellcheck.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    # Set working directory to `$test_dir` so the shellchek's
+    # `--external-sources` argument works better.
+    pushd "$test_dir"
+        find './' -type f -name '*.sh' -print0 | xargs -0 -I '{}' sh -c "echo 'Testing: {}'; ${shellcheck_args[*]} '{}'"
+    popd
+}
+
+# Run shellcheck over this script - this is covered by checking whole test directory below
+# scheck_file "$scriptDir/shellcheck.sh"
```
Comment (jonasnick) [inline test/ci/shellcheck.sh:28]: Why add this then (including the scheck_file function)

File: test/ci/shellcheck.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# TODO: add possibility to run this script directly
```
Comment (jonasnick) [inline test/ci/shellcheck.sh:3]: I think ideally it would be possible to run shellcheck with our `run-tests.sh` script and include it in `all()`.

**Review by jonasnick — COMMENTED** [review]
Thanks a lot @seberm, this is very helpful! I skimmed the shellcheck'd code and ran a few of the scripts. Looks good.

**Comment by seberm** [comment]
Hello @jonasnick ,
thank you for your review! I've made the suggested changes and added the shellcheck into the `run-tests.sh`. Let me know if everything works as expected.

**Comment by seberm** [comment]
Hello @jonasnick ,
I have rebased the changes against the current master.

File: test/ci/shellcheck.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+scheck_dir() {
+    local test_dir="$1"
+    echo "Shellchecking directory: ${test_dir}"
+
+    # Set working directory to `$test_dir` so the shellchek's
+    # `--external-sources` argument works better.
+    pushd "$test_dir"
+        find './' -type f -name '*.sh' -print0 | xargs -0 -I '{}' sh -c "echo 'Testing: {}'; ${shellcheck_args[*]} '{}'"
+    popd
```
Comment (jonasnick) [inline test/ci/shellcheck.sh:17]: nit: can you redirect the output of `pushd` and `popd` to `/dev/null`? This makes the output of `shellcheck.sh` much more comprehensible.

File: test/run-tests.sh [hunk 5]
```diff
@@ -312,19 +319,24 @@ examples() {
     (cd "$scriptDir/../examples" && nix-shell --run "$script")
 }
 
+shellcheck() {
+  . "$scriptDir/ci/shellcheck.sh"
+}
```
Comment (jonasnick) [inline test/run-tests.sh:324]: nit: Doesn't it make more sense to have shellcheck.sh in the `tests/` directory?

**Review by jonasnick — COMMENTED** [review]
Thanks @seberm. ACK mod nits. I want to merge this PR asap to avoid having to rebase it all the time. @erikarvstedt can you Concept ACK this?

**Review by jonasnick — DISMISSED** [review]
ACK c4e78281985aa53fec067ed778979aace956206e

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
Amazing, thanks a lot!

[Fixed branch](https://github.com/erikarvstedt/nix-bitcoin/commits/feature/shellcheck-for-helper-scripts-ea).

- Fix `start-bash-session.sh` which was broken due to shellcheck comments inserted into a
multi-line command.
- `run-tests.sh`: Fix rounding down of `memAvailableKiB`
- `run-tests.sh`: Add shellcheck binary to PATH for test `shellcheck`
- `shellcheck.sh`: Simplify, find all scripts in repo
- `.cirrus.yml`: Improve usage of the root flake
- `.cirrus.yml`: Use the actual Nix version from NixOS 22.05
- `.cirrus.yml`: Simplify test `shellcheck`

**Comment by erikarvstedt** [comment]
[Fixed branch](https://github.com/erikarvstedt/nix-bitcoin/commits/feature/shellcheck-for-helper-scripts-ea-2).

Thanks for fixing `fetch-release`.
Instead of renaming the file, I've now modified that test script to include files without extensions that have a
shell shebang.

**Comment by seberm** [comment]
Hello Erik,
you were quick :) !, thanks a lot for your review and the changes!

Hopefully I have rebased everything correctly :). Let me know if anything more is needed.

**Review by erikarvstedt — APPROVED** [review]
ACK 1a0df13d6c56d2120ecdb2027fc45c435f889cd4

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
Sorry, last fix: I've updated my branch above to pin shellcheck (and other binaries) used in the CI script to our version of nixpkgs.

**Comment by seberm** [comment]
> Sorry, last fix: I've updated my branch above to pin shellcheck (and other binaries) used in the CI script to our version of nixpkgs.

No problem.

**Review by erikarvstedt — APPROVED** [review]
ACK 91a03ce7d2a4d92edcb875938454da91b4178a21

**Review by jonasnick — COMMENTED** [review]
ACK 91a03ce7d2a4d92edcb875938454da91b4178a21

Thanks @seberm @erikarvstedt !

**Pull request merged**